src/tsconfig.ts

```diff
@@ -163,11 +163,23 @@ export function readFile (filename: string): Promise<any> {
         return reject(err)
       }
 
+      let obj
       try {
-        return resolve(parse(contents, filename))
+        obj = parse(contents)
       } catch (err) {
         return reject(err)
       }
+
+      if (obj.extends !== undefined) {
+        const filepath = filename.replace(path.basename(filename), '')
+        const extendsFilename = resolveSync(filepath, obj.extends) as string
```

Comment:
You can't do a synchronous operation in an async context, it defeats the purpose of supporting an async API.